src/lib/Libifl/tm.c

```diff
@@ -711,11 +712,24 @@ tm_attach(char *jobid, char *cookie, pid_t pid, tm_task_id *tid, char *host, int
 #endif
 
 	pbs_tcp_interrupt = 1;
-	tm_jobid = jobid;
-	tm_jobid_len = (tm_jobid != NULL) ? strlen(tm_jobid) : 0;
 
-	tm_jobcookie = cookie;
-	tm_jobcookie_len = (tm_jobcookie != NULL) ? strlen(tm_jobcookie) : 0;
+	tm_jobid_len = 0;
+	free(tm_jobid);
+	if (jobid && (*jobid != '\0')) {
+		tm_jobid = strdup(jobid);
+		if (!tm_jobid)
+			return TM_ESYSTEM;
+		tm_jobid_len = strlen(tm_jobid);
+	}
+
+	tm_jobcookie_len = 0;
+	free(tm_jobcookie);
+	if (cookie && (*cookie != '\0')) {
```

Comment:
shouldn't we need to add `tm_jobcookie= NULL;`  before this line?

Reply:
Yes.